scrapy/core/engine.py

```diff
@@ -84,6 +84,21 @@ def stop(self):
         dfd = self._close_all_spiders()
         return dfd.addBoth(lambda _: self._finish_stopping_engine())
 
+    def close(self):
+        """Close the execution engine gracefully.
+
+        If it has already been started, stop it. In all cases, close all spiders
+        and the downloader.
+        """
+        if self.running:
+            # Will close spiders and downloader
+            return self.stop()
+        elif self.open_spiders:
+            # Will close downloader
+            return self._close_all_spiders()
```

Comment:
Hehe I should reformulate the comments. They should read "Will **also** close spiders and downloader" etc.